src/java/ognl/ASTStaticField.java

```diff
@@ -89,15 +89,23 @@ public boolean isNodeConstant(OgnlContext context)
                 result = true;
             } else
             {
-                Field f = OgnlRuntime.getField(c, fieldName);
-                if (f == null) {
-                    throw new NoSuchFieldException(fieldName);
+                final int fModifiers;
+                Field f;
+
+                try {
+                    f = c.getField(fieldName);  // Public fields checked first (direct)
+                } catch (NoSuchFieldException nsfe) {
+                    f = OgnlRuntime.getField(c, fieldName);  // Non-public fields checked (access controlled)
+                    if (f == null) {
+                        throw new NoSuchFieldException(fieldName);
+                    }
                 }
+                fModifiers = f.getModifiers();
 
-                if (!Modifier.isStatic(f.getModifiers()))
```

Comment:
Why change this?

Reply:
Hello @aleksandr-m .
It was just a minor optimization (equivalent to one already used in `OgnlRuntime`) to replace _two calls_ to `f.getModifiers()`:
.. if (!Modifier.isStatic(f.getModifiers())) / result = Modifier.isFinal(f.getModifiers());
with a _single call/assignment_ for the field modifiers `fModifiers = f.getModifiers()`: 
.. if (!Modifier.isStatic(fModifiers)) / result = Modifier.isFinal(fModifiers).

Reply:
`getModifiers()` is just a getter it is ok to use it directly. If you want to keep `fModifiers` then declaration and assignment can be moved to them same line, currently it looks very odd.

Reply:
Hello @aleksandr-m .
Thanks for the feed-back and suggestion.
The declaration/assignment for `fModifiers` has been moved to the same line in both modules (and the declaration of one variable moved closer to its usage point).  Hopefully it looks a little cleaner now ?